src/plugins/es_ui_shared/public/request/use_request.helpers.tsx

```diff
@@ -0,0 +1,181 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { mount, ReactWrapper } from 'enzyme';
+import sinon from 'sinon';
+
+import { HttpSetup, HttpFetchOptions } from '../../../../../src/core/public';
+import { SendRequestConfig, SendRequestResponse } from './send_request';
+import { useRequest, UseRequestResponse } from './use_request';
+
+export interface UseRequestHelpers {
+  advanceTime: (ms: number) => Promise<void>;
+  completeRequest: () => Promise<void>;
+  hookResult: UseRequestResponse;
+  getSendRequestSpy: () => sinon.SinonStub;
+  setupSuccessRequest: (overrides?: {}) => void;
+  getSuccessResponse: () => SendRequestResponse;
+  setupErrorRequest: (overrides?: {}) => void;
+  getErrorResponse: () => SendRequestResponse;
+  setErrorResponse: () => void;
+  setupErrorWithBodyRequest: (overrides?: {}) => void;
+  getErrorWithBodyResponse: () => SendRequestResponse;
+}
+
+// Each request will take 1s to resolve.
+export const REQUEST_TIME = 1000;
+
+const successRequest: SendRequestConfig = { method: 'post', path: '/success', body: {} };
+const successResponse = { statusCode: 200, data: { message: 'Success message' } };
+
+const errorValue = { statusCode: 400, statusText: 'Error message' };
+const errorRequest: SendRequestConfig = { method: 'post', path: '/error', body: {} };
+const errorResponse = { response: { data: errorValue } };
+
+const errorWithBodyRequest: SendRequestConfig = {
+  method: 'post',
+  path: '/errorWithBody',
+  body: {},
+};
+const errorWithBodyResponse = { body: errorValue };
+
+export const createUseRequestHelpers = (): UseRequestHelpers => {
+  // The behavior we're testing involves state changes over time, so we need finer control over
+  // timing.
+  jest.useFakeTimers();
+
+  const flushPromiseJobQueue = async () => {
+    // See https://stackoverflow.com/questions/52177631/jest-timer-and-promise-dont-work-well-settimeout-and-async-function
+    await Promise.resolve();
+  };
+
+  const completeRequest = async () => {
+    await act(async () => {
+      jest.runAllTimers();
+      await flushPromiseJobQueue();
+    });
+  };
+
+  const advanceTime = async (ms: number) => {
```

Comment:
If we were to set a default here, you'd have to be familiar with the `advanceTime` implementation to know what that default is in order to read the code where it's called and understand what's going on. I find it easier to reason about the code by reading `advanceTime(REQUEST_TIME)` rather than `advanceTime()` and then recalling, "This will default to `REQUEST_TIME` in the implementation".